src/Lean/Elab/Tactic/GuardMsgs.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2023 Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller
+-/
+import Lean.Server.CodeActions.Attr
+-- import Std.Lean.Position
+
+/-! `#guard_msgs` command for testing commands
+
+This module defines a command to test that another command produces the expected messages.
+See the docstring on the `#guard_msgs` command.
+-/
+
+open Lean Parser.Tactic Elab Command
+
+namespace Std.Tactic.GuardMsgs
+
+/-- Gives a string representation of a message without source position information.
+Ensures the message ends with a '\n'. -/
+private def messageToStringWithoutPos (msg : Message) : IO String := do
+  let mut str ← msg.data.toString
+  unless msg.caption == "" do
+    str := msg.caption ++ ":\n" ++ str
+  match msg.severity with
+  | MessageSeverity.information => str := "info: " ++ str
+  | MessageSeverity.warning     => str := "warning: " ++ str
+  | MessageSeverity.error       => str := "error: " ++ str
```

Comment:
If `str` begins with a newline, then this ends up with trailing whitespace that has a tendency to get auto-deleted by editors. Should the space be suppressed if `str` begins with `\n`?